tests/Unit/Services/Radio/RadioStreamConnectorTest.php

```
@@ -0,0 +1,48 @@
+        $network = Mockery::mock(Network::class);
+        $network->expects('resolveUrlToPublicIps')->with('http://127.0.0.1/stream')->andReturn([]);
+
+        $connector = new RadioStreamConnector($network);
+
+        self::assertFalse($connector->connect('http://127.0.0.1/stream'));
+    }
+
+    #[Test]
+    public function connectsToTheValidatedAddressRatherThanResolvingAgain(): void
+    {
+        $server = stream_socket_server('tcp://127.0.0.1:0', $errorCode, $errorMessage);
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$errorCode'. (undefined)

(UnusedLocalVariable)